MediaBrowser.Model/Extensions/ContainerExtensions.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using Jellyfin.Extensions;
+
+namespace MediaBrowser.Model.Extensions;
+
+/// <summary>
+/// Defines the <see cref="ContainerExtensions"/> class.
+/// </summary>
+public static class ContainerExtensions
+{
+    /// <summary>
+    /// Compares two containers, returning true if an item in <paramref name="inputContainer"/> does not exist
+    /// in <paramref name="profileContainers"/>.
+    /// </summary>
+    /// <param name="profileContainers">The comma-delimited string being searched.
+    /// If the parameter begins with the - character, the operation is reversed.</param>
+    /// <param name="inputContainer">The comma-delimited string being matched.</param>
+    /// <returns>The result of the operation.</returns>
+    public static bool ContainsContainer(this string? profileContainers, string? inputContainer)
+    {
+        var isNegativeList = inputContainer == string.Empty;
+        if (profileContainers != null && profileContainers.StartsWith('-'))
+        {
+            isNegativeList = true;
+            profileContainers = profileContainers[1..];
+        }
+
+        return ContainsContainer(profileContainers, isNegativeList, inputContainer);
+    }
+
+    /// <summary>
+    /// Compares two containers, returning true if an item in <paramref name="inputContainer"/> does not exist
+    /// in <paramref name="profileContainers"/>.
+    /// </summary>
+    /// <param name="profileContainers">The comma-delimited string being searched.
+    /// If the parameter begins with the - character, the operation is reversed.</param>
+    /// <param name="inputContainer">The comma-delimited string being matched.</param>
+    /// <returns>The result of the operation.</returns>
+    public static bool ContainsContainer(this string? profileContainers, ReadOnlySpan<char> inputContainer)
+    {
+        var isNegativeList = inputContainer.IsEmpty;
+        if (profileContainers != null && profileContainers.StartsWith('-'))
+        {
+            isNegativeList = true;
+            profileContainers = profileContainers[1..];
+        }
+
+        return ContainsContainer(profileContainers, isNegativeList, inputContainer);
+    }
+
+    /// <summary>
+    /// Compares two containers, returning <paramref name="isNegativeList"/> if an item in <paramref name="inputContainer"/>
+    /// does not exist in <paramref name="profileContainers"/>.
+    /// </summary>
+    /// <param name="profileContainers">The comma-delimited string being searched.</param>
+    /// <param name="isNegativeList">The boolean result to return if a match is not found.</param>
+    /// <param name="inputContainer">The comma-delimited string being matched.</param>
+    /// <returns>The result of the operation.</returns>
+    public static bool ContainsContainer(this string? profileContainers, bool isNegativeList, string? inputContainer)
```

Comment:
Why do you have this overload? A string is automatically cast to a ReadOnlySpan

Reply:
After looking further into it: This overload is required because the input of the Span overload is not nullable. And apparently we have different behaviour for `null` and empty string: empty string allows all containers, null none.